wicketd-client/src/lib.rs

```diff
@@ -33,20 +33,16 @@ progenitor::generate_api!(
         RotInventory = { derives = [ PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize]},
         RotSlot = { derives = [ PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize]},
         ImageVersion = { derives = [ PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize]},
-        UpdateComponent = { derives = [ Copy, PartialEq, Eq, PartialOrd, Ord ] },
     },
     replace = {
         Duration = std::time::Duration,
-        ProgressEventForGenericSpec = update_engine::events::ProgressEvent<update_engine::NestedSpec>,
+        EventReportForWicketdEngineSpec = wicket_common::update_events::EventReport,
+        StepEventForWicketdEngineSpec = wicket_common::update_events::StepEvent,
+        ProgressEventForWicketdEngineSpec = wicket_common::update_events::ProgressEvent,
```

Comment:
I'll chalk this up to general unease about `replace` from the conversations you and Adam had previously, but: what are we gaining from these replacements? I don't see any method calls on the replaced types, for example, but maybe I'm missing it?

Reply:
Not at the moment, but I have work in progress to use `EventBuffer` within wicket, which requires that these are the wicket types.

I'll ask for a re-review after ensuring that that direction holds.